lib/screens/surveydetails/widget/answer_multi_choice_widget.dart

```diff
@@ -0,0 +1,118 @@
+import 'package:flutter/material.dart';
+import 'package:lydiaryanfluttersurvey/model/ui/question_ui_model.dart';
+import 'package:lydiaryanfluttersurvey/resources/dimensions.dart';
+
+class AnswerMultiChoiceWidget extends StatefulWidget {
+  final QuestionUiModel question;
+  final Function(List<String>) onChecked;
+
+  const AnswerMultiChoiceWidget({
+    Key? key,
+    required this.question,
+    required this.onChecked,
+  }) : super(key: key);
+
+  @override
+  State<AnswerMultiChoiceWidget> createState() =>
+      _AnswerMultiChoiceWidgetState();
+}
+
+class _AnswerMultiChoiceWidgetState extends State<AnswerMultiChoiceWidget> {
+  final List<String> _answers = [];
+  final List<int> _checkedIndex = [];
+
+  void _onCheckboxChecked(
+    bool? isChecked,
+    int index,
+    String text,
```

Comment:
We can replace two params with one 😇 
```suggestion
    String id,
```